prism_compile.c

```diff
@@ -7442,7 +7442,15 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
         if (PM_NODE_FLAG_P(node, PM_NODE_FLAG_STATIC_LITERAL)) {
             if (!popped) {
                 VALUE symbol = pm_static_literal_value(iseq, node, scope_node);
-                PUSH_INSN1(ret, location, putobject, symbol);
+
+                if (PM_NODE_FLAG_P(node, PM_INTERPOLATED_STRING_NODE)) {
```

Comment:
This one I'm not so sure about. `PM_NODE_FLAG_P` boils down to:

```c
node->flags & PM_INTERPOLATED_STRING_NODE
```

`PM_INTERPOLATED_STRING_NODE` is part of the node type enum. It happens to be 85, which is `1010101`. So I think this is effectively checking if the newline flag (the first common flag) is set. 

All of that is to say, I don't think this is quite the solution we're going for.